Services/WishListServices.cs

```diff
@@ -0,0 +1,312 @@
+using Orchard.Core.Title.Models;
+using Orchard.DisplayManagement;
+using Orchard.Environment.Extensions;
+using Orchard.Localization;
+using Orchard.Security;
+using Orchard.UI.Notify;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nwazet.Commerce.Services {
+    [OrchardFeature("Nwazet.WishLists")]
+    public class WishListServices : IWishListServices {
+        private readonly IContentManager _contentManager;
+        private readonly dynamic _shapeFactory;
+        private readonly IEnumerable<IProductAttributesDriver> _attributesDrivers;
+        private readonly INotifier _notifier;
+        private readonly IEnumerable<IWishListExtensionProvider> _wishListExtensionProviders;
+
+        public WishListServices(
+            IContentManager contentManager,
+            IShapeFactory shapeFactory,
+            IEnumerable<IProductAttributesDriver> attributesDrivers,
+            INotifier notifier,
+            IEnumerable<IWishListExtensionProvider> wishListExtensionProviders) {
+
+            _contentManager = contentManager;
+            _shapeFactory = shapeFactory;
+            _attributesDrivers = attributesDrivers;
+            _notifier = notifier;
+            _wishListExtensionProviders = wishListExtensionProviders;
+
+            T = NullLocalizer.Instance;
+        }
+
+        public Localizer T { get; set; }
+
+        private string DefaultWishListTitle {
+            get { return T("My WishList").Text; }
+        }
+
+        public IEnumerable<WishListListPart> GetWishLists(IUser user, int max = 0) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+
+            var wishLists = new List<WishListListPart>();
+            wishLists.Add(GetDefaultWishList(user));
+            var wishListsQuery = _contentManager.Query<WishListListPart, WishListListPartRecord>()
+               .Where(wllp => !wllp.IsDefault) //the default wishlist is already selected
+               .Join<CommonPartRecord>()
+               .Where(cpr => cpr.OwnerId == user.Id) //must match owner
+               .Join<TitlePartRecord>()
+               .OrderBy(p => p.Title);
+            if (max > 0) {
+                wishLists.AddRange(wishListsQuery.Slice(max - 1)); //-1, because the default wishlist is there already
+            } else {
+                wishLists.AddRange(wishListsQuery.List());
+            }
+
+            return wishLists;
+        }
+
+        public WishListListPart GetDefaultWishList(IUser user) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+            //get default wishlist
+            var wishList = _contentManager.Query<WishListListPart, WishListListPartRecord>()
+                .Where(wllp => wllp.IsDefault)
+                .Join<CommonPartRecord>()
+                .Where(cpr => cpr.OwnerId == user.Id) //match owner
+                .Slice(1).FirstOrDefault();
```

Comment:
I am not aware of `IContentQuery` having its own `FirstOrDefault` method, hence I am using `slice` because I only really want one result, and then I take the first item in the `IEnumerable` that returns.
Is that extension actually somewhere?